qa/evil-tests/src/test/java/org/elasticsearch/plugins/InstallPluginCommandTests.java

```diff
@@ -457,6 +459,31 @@ public void testConfigConflict() throws Exception {
         assertInstallCleaned(env);
     }
 
+    public void testMissingDescriptor() throws Exception {
+        Environment env = createEnv();
+        Path pluginDir = createTempDir();
+        Files.createFile(pluginDir.resolve("fake.yml"));
+        String pluginZip = writeZip(pluginDir, PathUtils.get("elasticsearch"));
+        UserError e = expectThrows(UserError.class, () -> {
+            installPlugin(pluginZip, env);
+        });
+        assertTrue(e.getMessage(), e.getMessage().contains("Could not find plugin descriptor 'plugin-descriptor.properties' in plugin zip"));
+        assertInstallCleaned(env);
+
+    }
+
+    public void testMissingDirectory() throws Exception {
+        Environment env = createEnv();
+        Path pluginDir = createTempDir();
+        Files.createFile(pluginDir.resolve(PluginInfo.ES_PLUGIN_PROPERTIES));
+        String pluginZip = writeZip(pluginDir, PathUtils.get(""));
+        UserError e = expectThrows(UserError.class, () -> {
+            installPlugin(pluginZip, env);
+        });
+        assertTrue(e.getMessage(), e.getMessage().contains("Could not find plugin descriptor 'plugin-descriptor.properties' in plugin zip"));
```

Comment:
If the directory is missing, I think that should be an error on its own, separate from the plugin descriptor missing? You could do this in the unzip code by using a flag to find if any files were found with the elasticsearch prefix?